lib/sidekiq_unique_jobs/queue_lock_timeout_calculator.rb

```diff
@@ -0,0 +1,25 @@
+module SidekiqUniqueJobs
+  class QueueLockTimeoutCalculator
+    include TimeoutCalculator
+
+    def self.for_item(item)
+      new(item)
+    end
+
+    def initialize(item)
+      @item = item
+    end
+
+    def seconds
+      time_until_scheduled + queue_lock_expiration
+    end
+
+    def queue_lock_expiration
```

Comment:
could we make these two methods the same so that it quacks like a `TimeoutCalculator` that responds to a `calculator.lock_expiration` instead? Feels overly specific with the run_lock_expiration method when the class already conveys that and we don't really care much about  this. Maybe even just putting it in seconds if that is what is used in the end. Feels like we have a few methods too many in these classes.

```ruby
module SidekiqUniqueJobs
  class TimeoutCalculator
    def self.for_item(item)
      new(item)
    end

    def initialize(item)
      @item = item
    end

    def time_until_scheduled
      scheduled = item[AT_KEY]
      return 0 unless scheduled
      (Time.at(scheduled) - Time.now.utc).to_i
    end

    def seconds
      raise NotImplementedError
    end

    def worker_class_queue_lock_expiration
      worker_class_expiration_for QUEUE_LOCK_TIMEOUT_KEY
    end

    def worker_class_run_lock_expiration
      worker_class_expiration_for RUN_LOCK_TIMEOUT_KEY
    end

    def worker_class
      @worker_class ||= SidekiqUniqueJobs.worker_class_constantize(item[CLASS_KEY])
    end

    private

    def worker_class_expiration_for(key)
      return unless worker_class.respond_to?(:get_sidekiq_options)
      worker_class.get_sidekiq_options[key]
    end

    attr_reader :item
  end
end

module SidekiqUniqueJobs
  class RunLockTimeoutCalculator < TimeoutCalculator
    def seconds
      @seconds ||= (
        worker_class_run_lock_expiration ||
        SidekiqUniqueJobs.config.default_run_lock_expiration
      ).to_i
    end
  end
end

module SidekiqUniqueJobs
  class QueueLockTimeoutCalculator
    def seconds
      queue_lock_expiration + time_until_scheduled
    end

    def queue_lock_expiration
      @queue_lock_expiration ||=
        (
          worker_class_queue_lock_expiration ||
          SidekiqUniqueJobs.config.default_queue_lock_expiration
        ).to_i
    end
  end
end
```

If I am not completely off with all the flu, fever, coughing and sneezing that should do pretty much the same thing as your code but is a lot less to maintain :)